pkg/kubelet/apis/podresources/server_v1_test.go

```diff
@@ -485,6 +518,185 @@ func TestAllocatableResources(t *testing.T) {
+		},
+	}
+
+	containers := []v1.Container{
+		{
+			Name: containerName,
+		},
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: podNamespace,
+			UID:       podUID,
+		},
+		Spec: v1.PodSpec{
+			Containers: containers,
+		},
+	}
+
+	pluginCDIDevices := []*podresourcesapi.CDIDevice{{Name: "dra-dev0"}, {Name: "dra-dev1"}}
+	draDevs := []*podresourcesapi.DynamicResource{
+		{
+			ClassName:      "resource-class",
+			ClaimName:      "claim-name",
+			ClaimNamespace: "default",
+			ClaimResources: []*podresourcesapi.ClaimResource{{CdiDevices: pluginCDIDevices}},
+		},
+	}
+
+	for _, tc := range []struct {
+		desc             string
+		err              error
+		exist            bool
+		pod              *v1.Pod
+		devices          []*podresourcesapi.ContainerDevices
+		cpus             []int64
+		memory           []*podresourcesapi.ContainerMemory
+		dynamicResources []*podresourcesapi.DynamicResource
+		expectedResponse *podresourcesapi.GetPodResourcesResponse
+	}{
+		{
+			desc:             "pod not exist",
+			err:              fmt.Errorf("pod %s in namespace %s not found", podName, podNamespace),
+			exist:            false,
+			pod:              nil,
+			devices:          []*podresourcesapi.ContainerDevices{},
+			cpus:             []int64{},
+			memory:           []*podresourcesapi.ContainerMemory{},
+			dynamicResources: []*podresourcesapi.DynamicResource{},
+
+			expectedResponse: &podresourcesapi.GetPodResourcesResponse{},
+		},
+		{
+			desc:             "pod without devices",
+			err:              nil,
+			exist:            true,
+			pod:              pod,
+			devices:          []*podresourcesapi.ContainerDevices{},
+			cpus:             []int64{},
+			memory:           []*podresourcesapi.ContainerMemory{},
+			dynamicResources: []*podresourcesapi.DynamicResource{},
+			expectedResponse: &podresourcesapi.GetPodResourcesResponse{
+				PodResources: &podresourcesapi.PodResources{
+					Name:      podName,
+					Namespace: podNamespace,
+					Containers: []*podresourcesapi.ContainerResources{
+						{
+							Name:             containerName,
+							Devices:          []*podresourcesapi.ContainerDevices{},
+							DynamicResources: []*podresourcesapi.DynamicResource{},
+						},
+					},
+				},
+			},
+		},
+		{
+			desc:             "pod with devices",
+			err:              nil,
+			exist:            true,
+			pod:              pod,
+			devices:          devs,
+			cpus:             cpus,
+			memory:           memory,
+			dynamicResources: draDevs,
+			expectedResponse: &podresourcesapi.GetPodResourcesResponse{
+				PodResources: &podresourcesapi.PodResources{
+					Name:      podName,
+					Namespace: podNamespace,
+					Containers: []*podresourcesapi.ContainerResources{
+						{
+							Name:             containerName,
+							Devices:          devs,
+							CpuIds:           cpus,
+							Memory:           memory,
+							DynamicResources: draDevs,
+						},
+					},
+				},
+			},
+		},
```

Comment:
similarly to above, I think we want a test with both devices and dynamic resources

Reply:
can be postponed.